sycl/source/detail/graph_impl.cpp

```diff
@@ -118,9 +118,10 @@ std::shared_ptr<node_impl> graph_impl::add(
   // Copy deps so we can modify them
   auto Deps = Dep;
   // A unique set of dependencies obtained by checking kernel arguments
+  // for accessors
   std::set<std::shared_ptr<node_impl>> UniqueDeps;
   for (auto &Arg : Args) {
-    if (Arg.MType != sycl::detail::kernel_param_kind_t::kind_pointer) {
+    if (Arg.MType != sycl::detail::kernel_param_kind_t::kind_accessor) {
```

Comment:
Pushed a test, and from stepping through in a debugger it looks like there are some edges being created here. However, because of the wider accessor issue the new test ultimately segfaults on graph enqueue, so can't say with confidence that this works (but it's still more correct than the previous USM edges)

Reply:
Ah my mistake, sorry! I forgot that changed.